ISPN-12353 REST allow backup restore progress to be submitted and polled (#8714)
infinispan/infinispan: documentation/src/main/asciidoc/topics/ref_rest_clusters.adoc

[id='rest_v2_restore_cluster']
= Restoring {brandname} Cluster Resources
Apply resources in a backup archive to restore {brandname} clusters. A `202`

> response indicates that the restore request has been accepted for processing.

Reviewer: ```suggestion
response indicates that the restore request is accepted for processing.
```

<zip-bytes>
--5ec9bc07-f069-4662-a535-46069afeda32--
----

[id='rest_v2_restore_cluster_list']
= Listing Restores
Retrieve the names of all restore requests that are in progress, completed, or
failed.

[source,options="nowrap",subs=attributes+]
----
GET /rest/v2/cluster/restores

> ----

Author: Actually, looking at this more, the only endpoints that actually return a value are `/backups` and `/restores` which are simply the names available, as well as the GET for `/backups/{backupName}` but that simply returns a `application/zip` payload which is documented. So I'm not sure there is anything to add?

> ----

Reviewer 2: so only pasting an example output or describing the format (I assume it's a JSON array?) is enough

> ----

Author: Sure, added. FYI this is missing in a lot of the REST examples from what I could see.

[id='rest_v2_restore_cluster_head']
= Checking Restore Progress
Verify that a restore operation is complete. A `200` response indicates the

> restore operation has completed. A `202` response indicates the backup operation is

Reviewer: ```suggestion
restore operation is complete. A `202` response indicates the backup operation is
```

in progress.

[source,options="nowrap",subs=attributes+]
----

> HEAD /rest/v2/cluster/restores/{restoreName}

Author: Purely for tracking purposes at the minute. I'll update the docs to clarify this.

----

[id='rest_v2_restore_cluster_delete']
= Deleting Restore Requests
Remove restore requests from the server. This does not remove any restored content, it
simply removes all meta information associated with a given request, therefore allowing

> subsequent restore requests to re-use a past request name.

Reviewer: Same as the previous comment. Maybe we should change the heading as well? 

= Deleting Request Metadata 
Remove metadata for restore requests from the server. This action removes all metadata associated with restore requests but does not delete any restored content. If you delete the request metadata, you can use the request name to perform subsequent restore operations.

Author: I like it, I have used your suggestion but with the title changed to "Deleting Restore Metadata"

A `204` response indicates that the  restore request has been deleted. A `202` response

> indicates that the restore operation is in progress but will be deleted when the operation

Reviewer: ```suggestion
indicates that the restore operation is in progress and will be deleted when the operation
```
